### ISSUE TITLE
[Product Preview] Product preview displays "Coming Soon" banner on free trial stores

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-ios#10137]
We don't fully authenticate product previews, which prevents previews from working on Woo Express free trial stores:

> I have a test store in a just-started state, and I created a product on iOS. I filled in a title and description, on a whim I tapped Preview before publishing.
> 
> Because I hadn’t sent my site live, all I saw was a Coming Soon screen.
> 
> Can we please somehow show a product preview on a non-live site?

Product details|Preview
-|-
![Image](https://user-images.githubusercontent.com/8658164/250052253-8b7125c7-083b-417f-996f-6248db3756cf.png)|![Image](https://user-images.githubusercontent.com/8658164/250052280-f6bd9447-4beb-424e-92b4-2b409dc625f2.png)

Raised here: p1688091479085639-slack-C02KUCFCSFP

**Comment by hafizrahman** [comment]
During maintenance day I tried attempting to solve this, I've pushed the branch `try/10137-preview-when-coming-soon` with my latest code.

The solution used `AuthenticatedWebViewController` for opening the preview. Sadly, it's still not working.

During debugging I can confirm that the WebView did go to `https://wordpress.com/wp-login.php` first, did a login, then tried to open the preview permalink. But it still opened the Coming Soon page.

I also tried substituting the preview permalink with internal WordPress.com page (`https://wordpress.com/me`) and it did open properly, which indicated that the login was successful. I wonder if something else is needed to bypass the Coming Soon page.

I'm running out of time so I'm leaving the branch as-is, in case it's helpful to someone in the future who's tackling this.

**Comment by itsmeichigo** [comment]
@hafizrahman I tried opening one of my WooExpress stores on my desktop browser and I'm seeing the same thing even though I'm authenticated with my WPCom credentials already. I had to tap the Log In button on the page to see the demo site. I wonder if this is an issue with the web and not something we can fix from the app.

**Comment by hafizrahman** [comment]
@itsmeichigo interesting! Testing preview in the Web does work for me. I also played around with the permalink URL format, making sure the app's webview was opening the exact same URL that worked in Chrome, but it's still not bypassing.

**Comment by rachelmcr** [comment]
> During debugging I can confirm that the WebView did go to https://wordpress.com/wp-login.php first, did a login, then tried to open the preview permalink. But it still opened the Coming Soon page.

@hafizrahman Are you sure the login was successful? I tested with your branch and from what I'm seeing, the login doesn't actually happen. (I haven't investigated further to see what's actually happening, though.)

I then tried going to the Menu tab > WooCommerce Admin to open that webview. I saw the login screen, logged in, and after that point all the webviews (including product previews) were logged in and the previews worked for me.

**Comment by hafizrahman** [comment]
> Are you sure the login was successful? I tested with your branch and from what I'm seeing, the login doesn't actually happen.

@rachelmcr I had assumed so because when I modified the branch code to use wordpress.com/me as the URL to open, it opened correctly:

https://github.com/woocommerce/woocommerce-ios/assets/266376/e929ba32-f76d-4c18-b266-4478aa73ca09

**Comment by rachelmcr** [comment]
I think the issue is that the login to WordPress.com works (as you found with your test) but the SSO authentication for the site's wp-admin isn't complete. I tested that theory on the web:

1. Copied a preview link from my Woo Express store.
2. Visited WordPress.com logged out.
3. Logged in to WordPress.com.
4. Immediately tried to open the preview link (without visiting wp-admin first).

I got the same results as the app, where I see the "Coming Soon" message. When I try to then visit wp-admin, I see the SSO login prompt for the site, so I might be logged in to WordPress.com but I'm not authenticated in wp-admin.
